app/src/main/java/pl/edu/kasprzak/tictactoe/Board.java

```diff
@@ -1,17 +1,74 @@
 package pl.edu.kasprzak.tictactoe;
 
+import android.view.MotionEvent;
+import android.view.View;
 import android.widget.Button;
+import android.widget.Toast;
 
 public class Board {
     private Button buttons[][];
     private MainActivity activity;
+    private boolean gameOver = false;
+    String winner;
+
     public Board(MainActivity activity) {
         this.activity = activity;
         buttons = new Button[3][3];
     }
-    public void setButton(Button button, int x, int y) {
+
+    public void setButton(Button button, int x, int y, Board board) {
         buttons[x][y] = button;
-        button.setOnClickListener(new ButtonOnClick(button));
+        button.setOnClickListener(new ButtonOnClick(button, board));
```

Comment:
setButton(Button button, int x, int y, Board board)
tutaj nie musieliśmy dodawać argumentu board wystarczy użyć this:
new ButtonOnClick(button, board)
new ButtonOnClick(button, this)
Ta metoda znajduje się w klasie board. I ten this zwraca właśnie klasę w której się aktualnie znajdujemy czyli w tym przypadku board.